winter/routing/route.py

```diff
@@ -1,21 +1,56 @@
+import re
 import typing
 
-import dataclasses
 import uritemplate
+from uritemplate import URITemplate
 
 from ..core import ComponentMethod
-from ..core import cached_property
-from ..http import MediaType
+from .query_parameters.map_query_parameter_annotation import MapQueryParameterAnnotation
+from .query_parameters.query_parameter import QueryParameter
+
+_remove_query_params_regexp = re.compile(r'{\?[^}]*}')
 
 
-@dataclasses.dataclass(frozen=True)
 class Route:
-    http_method: str
-    url_path: str
-    method: ComponentMethod
-    produces: typing.Tuple[MediaType] = None
-    consumes: typing.Tuple[MediaType] = None
-
-    @cached_property
-    def path_variables(self):
+
+    def __init__(
+        self,
+        http_method: str,
+        url_path_with_query_parameters: str,
+        method: ComponentMethod,
```

Comment:
I have removed produces and consumes, because now we do not use it
(there was an issue that only 4 param allowed from codeclimate)

Reply:
It is quite tricky to use dataclasses. We can use more than 4 parameters in init with dataclasses